src/frontend/apps/lti_site/components/VideoWidgetProvider/widgets/UploadSubtitles/ToggleSubtitlesAsTranscript/index.tsx

```diff
@@ -92,22 +84,58 @@ export const ToggleSubtitlesAsTranscript = () => {
     });
   }, [video.should_use_subtitle_as_transcript, videoMutation]);
 
-  // if there is no timed text track, do not display this component.
-  if (timedTextTracks.length === 0) {
-    return null;
-  }
+  const transcripts = useMemo(
+    () =>
+      timedTextTracks
+        .filter((track) => track.is_ready_to_show)
+        .filter((track) => timedTextMode.TRANSCRIPT === track.mode),
+    [timedTextTracks],
+  );
 
-  const transcripts = timedTextTracks
-    .filter((track) => track.is_ready_to_show)
-    .filter((track) => timedTextMode.TRANSCRIPT === track.mode);
+  const subtitles = useMemo(
+    () =>
+      timedTextTracks
+        .filter((track) => track.is_ready_to_show)
+        .filter((track) => timedTextMode.SUBTITLE === track.mode),
+    [timedTextTracks],
+  );
 
-  const subtitles = timedTextTracks
-    .filter((track) => track.is_ready_to_show)
-    .filter((track) => timedTextMode.SUBTITLE === track.mode);
+  useEffect(() => {
+    setToggleUseTranscript(video.should_use_subtitle_as_transcript);
+  }, [video.should_use_subtitle_as_transcript]);
+
+  useEffect(() => {
+    if (subtitles.length === 0) {
+      if (!disabledToggle) {
+        setDisabledToggle(true);
+        if (video.should_use_subtitle_as_transcript) {
+          onToggleChange();
+        }
+      }
+    }
 
-  if (transcripts.length > 0 || subtitles.length === 0) {
-    return null;
-  }
+    if (transcripts.length > 0) {
+      if (!disabledToggle) {
+        setDisabledToggle(true);
+        if (video.should_use_subtitle_as_transcript) {
+          onToggleChange();
+        }
+      }
+    }
+  }, [
+    subtitles,
+    transcripts,
```

Comment:
I think you should create another useEffect just for `transcripts`.
As it is now, if subtitles change and length equals 0 and transcript.lenght > 0, you will call onToggleChange 2 times on a row.